# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/wifi/moz.build
# Commit: af2b512cb5ac
# Full Hash: af2b512cb5ac7a2b53224f5065f7d89fd8a503ed
# Author: David Parks <daparks@mozilla.com>
# Date: 2023-05-13 21:42:25
# Regressor Bug: 1810421
# File Overlap Count: 2
# Description:
#   Bug 1810421: Refactor wifi scanning to consolidate generic code r=necko-reviewers,kershaw,valentin a=reland CLOSED TREE
#   
#   Reorganizes the wifi-scanning code and makes more of it platform-generic
#   to ease the transition from polling the wifi to usually scanning only on
#   network changes.  This is essentially just moving files/code around and
# ==============================================================================

diff -r 3bb8a8b46f3c -r af2b512cb5ac netwerk/wifi/moz.build
--- a/netwerk/wifi/moz.build	Sat May 13 16:09:03 2023 +0000
+++ b/netwerk/wifi/moz.build	Sat May 13 00:36:00 2023 +0000
@@ -18,34 +18,44 @@
 ]
 
 if CONFIG["OS_ARCH"] == "Darwin":
-    UNIFIED_SOURCES += [
-        "nsWifiScannerMac.cpp",
+    SOURCES += [
+        "mac/MacWifiScanner.mm",
     ]
-    SOURCES += [
-        "osx_corewlan.mm",
+    LOCAL_INCLUDES += [
+        "mac",
     ]
 elif CONFIG["OS_ARCH"] in ("DragonFly", "FreeBSD"):
     UNIFIED_SOURCES += [
-        "nsWifiScannerFreeBSD.cpp",
+        "freebsd/FreeBsdWifiScanner.cpp",
+    ]
+    LOCAL_INCLUDES += [
+        "freebsd",
     ]
 elif CONFIG["OS_ARCH"] == "WINNT":
     UNIFIED_SOURCES += [
-        "nsWifiScannerWin.cpp",
-        "win_wifiScanner.cpp",
-        "win_wlanLibrary.cpp",
+        "win/WinWifiScanner.cpp",
+        "win/WlanLibrary.cpp",
+    ]
+    LOCAL_INCLUDES += [
+        "win",
     ]
 elif CONFIG["OS_ARCH"] == "SunOS":
     CXXFLAGS += CONFIG["GLIB_CFLAGS"]
     UNIFIED_SOURCES += [
-        "nsWifiScannerSolaris.cpp",
+        "solaris/SolarisWifiScanner.cpp",
     ]
-
-if CONFIG["NECKO_WIFI_DBUS"]:
+    LOCAL_INCLUDES += [
+        "solaris",
+    ]
+elif CONFIG["NECKO_WIFI_DBUS"]:
     UNIFIED_SOURCES += [
-        "nsWifiScannerDBus.cpp",
+        "dbus/DbusWifiScanner.cpp",
     ]
-
-if CONFIG["NECKO_WIFI_DBUS"]:
+    LOCAL_INCLUDES += [
+        "dbus",
+    ]
     CXXFLAGS += CONFIG["MOZ_DBUS_GLIB_CFLAGS"]
 
 FINAL_LIBRARY = "xul"
+
+include("/ipc/chromium/chromium-config.mozbuild")